packages/leancode_lint/lib/lints/missing_cleanup.dart

```diff
@@ -0,0 +1,374 @@
+        return;
+      }
+
+      final classNode = _getContainingClass(node);
+
+      if (classNode == null ||
+          !(_isWidgetClass(classNode) || _isStateOfWidget(classNode)) ||
+          _getDisposableMethodName(type) == null) {
+        return;
+      }
+
+      if (_isInReturnWidget(node)) {
+        reporter.atNode(node, code);
+      }
+    });
+  }
+
+  bool _isIgnoredInstance(InterfaceType type) => config.ignoredTypesCheckers
+      .any((checker) => checker.isExactly(type.element));
+
+  InterfaceType? _getFieldDeclarationType(FieldDeclaration field) {
+    if (field.fields.type?.type case final InterfaceType type) {
+      return type;
+    }
+    return switch (field.fields.variables.first.initializer?.staticType) {
+      final InterfaceType type => type,
+      _ => null,
+    };
+  }
+
+  bool _isFieldUsedByConstructor(
+    ClassDeclaration classNode,
+    FieldDeclaration node,
+  ) {
+    final constructorDeclaration = _getConstructorDeclaration(classNode);
+    if (constructorDeclaration == null) {
+      return false;
+    }
+
+    return _hasConstructorParameterOrInitializer(
+      constructorDeclaration,
+      node.fields.variables.first.name.lexeme,
+    );
+  }
+
+  bool _hasConstructorParameterOrInitializer(
+    ConstructorDeclaration constructorDeclaration,
+    String parameterName,
+  ) {
+    if (constructorDeclaration.parameters.parameters.any(
+      (parameter) => parameter.name?.lexeme == parameterName,
+    )) {
+      return true;
+    }
+
+    return constructorDeclaration.initializers
+        .whereType<ConstructorFieldInitializer>()
+        .any((initializer) => initializer.fieldName.name == parameterName);
+  }
+
+  ConstructorDeclaration? _getConstructorDeclaration(
+    ClassDeclaration classNode,
+  ) => classNode.members.whereType<ConstructorDeclaration>().firstOrNull;
+
+  ClassDeclaration? _getContainingClass(AstNode node) {
+    var classNode = node.parent;
+    while (classNode != null && classNode is! ClassDeclaration) {
+      classNode = classNode.parent;
+    }
+    return switch (classNode) {
+      final ClassDeclaration classNode => classNode,
+      _ => null,
+    };
+  }
+
+  String? _getDisposableMethodName(InterfaceType type) {
+    return type.methods2
+            .firstWhereOrNull(
+              (method) => config.cleanupMethods.contains(method.name3),
+            )
+            ?.name3 ??
+        type.element3.inheritedMembers.entries
+            .firstWhereOrNull(
+              (entry) =>
+                  config.cleanupMethods.contains(entry.key.name) &&
+                  entry.value.baseElement is MethodElement2,
+            )
+            ?.key
+            .name;
+  }
+
+  bool _isWidgetType(InterfaceType type) {
+    const widgetTypeChecker = TypeChecker.fromName(
+      'Widget',
+      packageName: 'flutter',
+    );
+
+    return widgetTypeChecker.isExactlyType(type) ||
+        widgetTypeChecker.isSuperTypeOf(type);
```

Comment:
Doesn't being exactly the type imply being a supertype? i.e. the check for isExactlyType is redundant?